cloudbridge/cloud/providers/azure/resources.py

```diff
@@ -0,0 +1,1701 @@
+
+    def iter_content(self):
+        """
+        Returns this object's content as an
+        iterable.
+        """
+        content_stream = self._provider.azure_client. \
+            get_blob_content(self._container.name, self._key.name)
+        if content_stream:
+            content_stream.seek(0)
+        return content_stream
+
+    def upload(self, data):
+        """
+        Set the contents of this object to the data read from the source
+        string.
+        """
+        try:
+            self._provider.azure_client.create_blob_from_text(
+                self._container.name, self.name, data)
+            return True
+        except AzureException as azureEx:
+            log.exception(azureEx)
+            return False
+
+    def upload_from_file(self, path):
+        """
+        Store the contents of the file pointed by the "path" variable.
+        """
+        try:
+            self._provider.azure_client.create_blob_from_file(
+                self._container.name, self.name, path)
+            return True
+        except AzureException as azureEx:
+            log.exception(azureEx)
+            return False
+
+    def delete(self):
+        """
+        Delete this object.
+
+        :rtype: bool
+        :return: True if successful
+        """
+        try:
+            self._provider.azure_client.delete_blob(
+                self._container.name, self.name)
+            return True
+        except AzureException as azureEx:
+            log.exception(azureEx)
+            return False
+
+    def generate_url(self, expires_in=0):
+        """
+        Generate a URL to this object.
+        """
+        return self._provider.azure_client.get_blob_url(
+            self._container.name, self.name)
+
+
+class AzureBucket(BaseBucket):
+    def __init__(self, provider, bucket):
+        super(AzureBucket, self).__init__(provider)
+        self._bucket = bucket
+
+    @property
+    def id(self):
+        return self._bucket.name
+
+    @property
+    def name(self):
+        """
+        Get this bucket's name.
+        """
+        return self._bucket.name
+
+    def get(self, key):
+        """
+        Retrieve a given object from this bucket.
+        """
+        try:
+            obj = self._provider.azure_client.get_blob(self.name, key)
+            return AzureBucketObject(self._provider, self, obj)
+        except AzureException as azureEx:
+            log.exception(azureEx)
+            return None
+
+    def list(self, limit=None, marker=None, prefix=None):
+        """
+        List all objects within this bucket.
+
+        :rtype: BucketObject
+        :return: List of all available BucketObjects within this bucket.
+        """
+        objects = [AzureBucketObject(self._provider, self, obj)
+                   for obj in
+                   self._provider.azure_client.list_blobs(
+                       self.name, prefix=prefix)]
+        return ClientPagedResultList(self._provider, objects,
```

Comment:
@madhugilla @VikramAIS This should be using a ServerPagedResultList instead of a ClientPagedResultList whenever possible. This may account for why the tests are running so slowly - since all objects are being retrieved before being filtered. Instead, the marker and maxresults parameters in list_blobs could be used to make this more efficient.